src/System.Runtime/src/System/HashCode.cs

```diff
@@ -0,0 +1,406 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    public struct HashCode
+    {
+#   if SYSTEM_HASHCODE_TESTVECTORS
+        private static readonly uint _seed = 0;
+#   else
+        private static readonly uint _seed = unchecked((uint)new Random().Next(int.MinValue, int.MaxValue));
+#   endif
+
+        private const uint Prime1 = 2654435761U;
+        private const uint Prime2 = 2246822519U;
+        private const uint Prime3 = 3266489917U;
+        private const uint Prime4 = 0668265263U;
+        private const uint Prime5 = 0374761393U;
+
+        private uint _v1, _v2, _v3, _v4;
+        private uint _queue1, _queue2, _queue3;
+        private uint _length;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Right, it should be supported by data that shows it makes difference.

Reply:
Last when I checked (with my implementations in Haschisch) it made a difference. It could easily be that (at best) some are redundant (as the JIT would inline anyways), but any single function call that remains will cause performance to drop. Also, the typical `.GetHashCode()` caller usually will not be inlined anyways.

Reply:
>supported by data

Just remembered that [there's an app for that.](https://marketplace.visualstudio.com/items?itemName=StephanZehetner.InliningAnalyzer)

Most of the methods run up against "Native estimate for function size exceeds threshold" (with exception to `MixEmptyState`). Writing to to arguments (all the mix methods) outright prevented inlining even with `AggressiveInlining`. Switch statements are a no-no. Curiously `Rol` doesn't get inlined, I thought it was supposed to be a JIT intrinsic?

The tool doesn't support generic methods at all. Everything else is inlined after [the latest batch of changes:](https://github.com/jcdickinson/corefx/commit/1775f538d2a030f457c4d67af5642d09a6566706)

![Inlined methods](https://user-images.githubusercontent.com/522465/32387609-3f710cbc-c082-11e7-8818-000a9592eee2.png)